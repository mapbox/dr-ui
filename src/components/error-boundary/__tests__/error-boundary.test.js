import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './error-boundary-test-cases.js';
import { mount } from 'enzyme';

describe('error-boundary', () => {
  describe(testCases.basic.description, () => {
    const testCase = testCases.basic;

    test('renders as expected', () => {
      const wrapper = renderer.create(testCase.element).toJSON();
      expect(wrapper).toMatchSnapshot();
    });

    test('no error', () => {
      const wrapper = mount(testCase.element);
      expect(wrapper.state()).toEqual(null);
    });
  });

  describe(testCases.oops.description, () => {
    const testCase = testCases.oops;

    test('renders as expected', () => {
      const wrapper = renderer
        .create(React.createElement(testCase.component, testCase.props))
        .toJSON();
      expect(wrapper).toMatchSnapshot();
    });

    test('error is caught', () => {
      const wrapper = mount(
        React.createElement(testCase.component, testCase.props)
      );
      const err = new Error(
        'BadFunction(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.'
      );
      expect(wrapper.state()).toEqual({ hasError: true, error: err });
    });
  });

  describe(testCases.oopsCustom.description, () => {
    const testCase = testCases.oopsCustom;

    test('renders as expected', () => {
      const wrapper = renderer
        .create(React.createElement(testCase.component, testCase.props))
        .toJSON();
      expect(wrapper).toMatchSnapshot();
    });

    test('error is caught', () => {
      const wrapper = mount(
        React.createElement(testCase.component, testCase.props)
      );
      const err = new Error(
        'BadFunction(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.'
      );
      expect(wrapper.state()).toEqual({ hasError: true, error: err });
    });
  });
});
