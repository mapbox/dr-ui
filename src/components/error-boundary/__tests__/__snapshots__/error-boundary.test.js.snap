// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error-boundary Basic renders as expected 1`] = `"This content does not trigger an error."`;

exports[`error-boundary Trigger error renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#fbe5e5",
      "color": "#9b3134",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-red-light round-full color-red flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-alert"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Something went wrong
    </div>
    <p>
      We sent the error to the team. Please refresh the page to try again.
    </p>
    <details>
      <summary
        className="cursor-pointer"
      >
        Error details
      </summary>
      <pre>
        <code>
          BadFunction(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.
        </code>
      </pre>
    </details>
  </div>
</div>
`;

exports[`error-boundary Trigger error with custom title and note contents renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#fbe5e5",
      "color": "#9b3134",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-red-light round-full color-red flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-alert"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Well this is embarassing
    </div>
    <p>
      This feature is having trouble loading, please try again later.
    </p>
    <details>
      <summary
        className="cursor-pointer"
      >
        Error details
      </summary>
      <pre>
        <code>
          BadFunction(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.
        </code>
      </pre>
    </details>
  </div>
</div>
`;
